### PR TITLE
discuss: NNS proposal draft

### PR DESCRIPTION
**Pull request by roman-kashitsyn**

### CODE REVIEW

File: PROPOSAL.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+# The ICRC-1 Fungible Token Standard
+
+## Introduction
+
+​​The ICRC-1 Fungible Token Standard is the product of the collaborative effort of the [Ledger & Tokenization working group](https://forum.dfinity.org/t/announcing-token-standard-as-topic-of-the-first-meeting-of-the-ledger-tokenization-working-group/11925/1), including members from the DFINITY foundation, InfinitySwap, ICPSwap, and the ORIGYN Foundation.
```
Comment (roman-kashitsyn) [inline PROPOSAL.md:5]:
```suggestion
​​The ICRC-1 Fungible Token Standard is the product of the collaborative effort of the [Ledger & Tokenization working group](https://forum.dfinity.org/t/announcing-token-standard-as-topic-of-the-first-meeting-of-the-ledger-tokenization-working-group/11925/1), including members from the DFINITY foundation, InfinitySwap, ICPSwap, ICDevs, and the ORIGYN Foundation.
```

File: PROPOSAL.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+* It is possible to integrate ICRC-1 endpoints into all major token ledger implementations, including the ICP ledger and most ledgers modeled after the [ERC-20](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/) token standard (see the [Two-component account structure](#two-component-account-structure) subsection for more detail).
+* The standard is known to have [a scalable and secure implementation](https://github.com/dfinity/ic/tree/1b2e15f17f8a2beaf28f802d675240f252585a30/rs/rosetta-api/icrc1) that will ship with the [SNS](https://medium.com/dfinity/how-the-service-nervous-system-sns-will-bring-tokenized-governance-to-on-chain-dapps-b74fb8364a5c).
+* The standard has a modular design allowing future extensions without the loss of backward compatibility (see the [Modularity](#modularity) subsection).
+
+## Comments on the arguments against this standard
+
+The Ledger & Tokenization working group collected feedback from its members and the community. Below are the common arguments against accepting ICRC-1 as a standard and the commentary from the working group.
+
+* **ICRC-1 does not provide a reasonable payment flow.**
```
Comment (levifeldman) [inline PROPOSAL.md:48]:
```suggestion
* **"ICRC-1 does not provide a reasonable payment flow."**
```

File: PROPOSAL.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+  The standard is intentionally conservative and purposely avoids features that have known security flaws or can compromise the correctness or scalability of implementation with the current state of the core platform.
+  Instead of cramming as much functionality as possible into a single API, the working group decided to include only the necessary and universal functionality into ICRC-1 and standardize additional payment flows as future extensions (see the [Modularity](#modularity) subsection).
+  Note that the [proposed interface](#the-application-programming-interface) is powerful enough to support a payment flow based on subaccounts.
+  All canisters smart contracts interacting with the ICP ledger rely on that flow.
+
+* **ICRC-1 provides no benefits over alternatives.**
```
Comment (levifeldman) [inline PROPOSAL.md:55]:
```suggestion
* **"ICRC-1 provides no benefits over alternatives."**
```

File: PROPOSAL.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+  ICRC-1 does not seek to replace existing standards but rather complement them, providing a unified interface for token transfers.
+  Most token ledgers should be able to become standard-compliant without having to throw away their transaction logs.
+
+* **It is too late to introduce a standard.**
```
Comment (levifeldman) [inline PROPOSAL.md:60]:
```suggestion
* **"It is too late to introduce a standard."**
```

File: PROPOSAL.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+  The core team is working on many features that can reshape DeFi on the Internet Computer.
+  Some working group members think it is too early to agree on a single standard for all purposes!
+  ICRC-1's extension mechanism allows the DeFi capabilities to grow as the platform evolves.
+  See the [Modularity](#modularity) subsection for more detail.
+
+* **The account addressing scheme involving subaccounts is unnecessarily complicated.**
```
Comment (levifeldman) [inline PROPOSAL.md:67]:
```suggestion
* **"The account addressing scheme involving subaccounts is unnecessarily complicated."**
```

**Review by levifeldman — CHANGES_REQUESTED** [review]

File: PROPOSAL.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+### Namespaces
+
+All ICRC-1 [methods](https://github.com/dfinity/ICRC-1#methods) have the `icrc1_` prefix.
+This feature allows existing smart contracts to implement ICRC-1 methods without introducing name clashes.
+
+### Unspecified transaction structure
+
+ICRC-1 _does not_ specify the structure of transactions or provide methods for fetching them.
+That deliberate design choice enables backward compatibility with existing token ledgers, such as the ICP ledger.
+The Ledger & Tokenization working group plans to work on an _optional_ core extension specifying the interface for fetching transactions.
```
Comment (levifeldman) [inline PROPOSAL.md:145]:
```suggestion
The Ledger & Tokenization working group plans to work on an _optional_ extension specifying the interface for fetching transactions.
```
The word 'core' and the word 'extension' are opposites and putting them in the same-phrase is contradictory.

File: PROPOSAL.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+The DFINITY Foundation plans to implement the ICRC-1 interface in the ICP ledger canister smart contract.
+Though ICRC-1 is compatible with the ICP ledger design, transferring funds to 28-byte ICP account identifiers will not be possible through the ICRC-1 interface.
+You will have to use the existing ICP ledger interface for such transfers.
+
+
+## Conclusion
+
+The [Ledger & Tokenization working group](https://forum.dfinity.org/t/announcing-token-standard-as-topic-of-the-first-meeting-of-the-ledger-tokenization-working-group/11925/1) proposes to adopt the [ICRC-1](https://github.com/dfinity/ICRC-1) token standard as a common interface for transferring fungible tokens on the Internet Computer.
+The standard does not seek to replace existing token ledgers but rather to serve as a base interface for accessing them.
+The working group agrees to implement, integrate, and promote the standard and work on its core extensions.
```
Comment (levifeldman) [inline PROPOSAL.md:181]:
```suggestion
The working group agrees to implement, integrate, and promote the standard and work on its optional extensions.
```
Make it clear that an extension is always optional and is not needed to implement the ICRC-1 standard.

File: PROPOSAL.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+
+### Modularity
+
+ICRC-1 is the _base_ standard designed for extension with follow-up specifications as new flows become practical.
+Many efficient flows require unique features of the IC platform, such as [Threshold ECDSA signatures](https://medium.com/dfinity/threshold-ecdsa-the-key-ingredient-behind-the-internet-computers-bitcoin-and-ethereum-cf22649b98a1), to become stable and well-understood.
+The standard defines the [icrc1_supported_standards](https://github.com/dfinity/ICRC-1#icrc1_supported_standards) method listing all standards the token ledger supports.
+This method will allow applications such as wallets and Rosetta API implementations to discover and take advantage of additional features in the future.
+
+The Ledger & Tokenization working group will continue working on a set of _core_ extensions recommended for all ledger implementations.
```
Comment (levifeldman) [inline PROPOSAL.md:160]:
```suggestion
The Ledger & Tokenization working group will continue working on a set of _optional_ extensions recommended for all ledger implementations.
```
The phrase: "core extensions" is one that contradicts itself by the definition of the word 'core' and of the word 'extension'. 'optional' makes it clear that the ICRC-1 standard stands on its own.

**Review by levifeldman — CHANGES_REQUESTED** [review]

File: PROPOSAL.md (hunk 8 above)
Comment (roman-kashitsyn) [reply PROPOSAL.md]: Good point, rephrased the sentence.

**Comment by roman-kashitsyn** [comment]
The proposal went live: https://dashboard.internetcomputer.org/proposal/74740

**Pull request closed without merging**